### PR TITLE
Improve NVVP service with regular expressions

### PR DESCRIPTION
**Pull request by davidbeckingsale**
😀

### CODE REVIEW

**Comment by davidbeckingsale** [comment]
@daboehme @DavidPoliakoff I'd appreciate you looking over this before a merge.

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -11,6 +11,7 @@ else()
 endif()
 
 include_directories(${PROJECT_BINARY_DIR})
+include_directories(${CMAKE_CURRENT_SOURCE_DIR})
```
Comment (davidbeckingsale) [inline src/CMakeLists.txt]: This is so we can do root-level includes.

File: src/common/filters/RegexFilter.h [hunk 2]
```diff
@@ -7,16 +7,40 @@
...
+
 class RegexFilter : public Filter<RegexFilter> {
+  private:
+    static std::string regex;
+    static cali::ConfigSet config;
+
   public:
+    static void initialize()
+    {
+      config = cali::RuntimeConfig::init("nvvp", s_configdata);
```
Comment (daboehme) [inline src/common/filters/RegexFilter.h]: Eventually the config needs to be provided by whoever instantiates a RegexFilter. We can leave it in as a hack for now.

Comment (davidbeckingsale) [inline src/common/filters/RegexFilter.h]: Agree!

File: src/services/nvvp/nvvp.cpp [hunk 3]
```diff
@@ -33,49 +33,61 @@
...
+      eventAttrib.message.ascii = name.c_str();
+      nvtx_ranges[name] = nvtxRangeStartEx(&eventAttrib);
     }
 
     static void endAction(Caliper* c, const Attribute& attr, const Variant& value){
-         CALIPER_NVVP_POP_RANGE
+      std::stringstream ss;
+      ss << attr.name() << "=" << value.to_string();
+      std::string name = ss.str();
+      nvtxRangeId_t r = nvtx_ranges[name];
```
Comment (daboehme) [inline src/services/nvvp/nvvp.cpp]: Add a check - this will fail whenever you use `set(); end();` instead of `begin(); end();`

Comment (davidbeckingsale) [inline src/services/nvvp/nvvp.cpp]: Okay, I'll fix this up.

**Pull request merged**